CMakeLists.txt

```diff
@@ -339,12 +339,10 @@ if (${ENABLE_PROFILE} AND HAVE_COMPILER_GNU_COMPAT)
 	link_libraries(-g -pg)
 endif()
 
-if (MINGW)
-	find_library(PTHREAD_LIBRARY NAMES pthread pthreadGC2 pthreadGC)
-elseif (PTHREAD_LIBRARY AND PTHREAD_INCLUDE_DIR)
+if (PTHREAD_LIBRARY AND PTHREAD_INCLUDE_DIR)
 	message(STATUS "Pthread library: ${PTHREAD_LIBRARY}")
 	message(STATUS "Pthread include dir: ${PTHREAD_INCLUDE_DIR}")
-elseif (WIN32)
+elseif (WIN32 AND NOT MINGW)
```

Comment:
This expression has a shortcut called `MICROSOFT`. Not sure if that was exactly the intent, but it means in short {Compiling on Windows and using Microsoft Visual Studio}.

I thought also that these names: pthread, pthreadGC2 and pthreadGC were added for a reason some time ago, it was a fix from community.

Reply:
Ok, thanks for pointing out the MICROSOFT shortcut but that hardly deserves revising the code I think. Maybe on next round we have to touch it.

`find_library(PTHREAD_LIBRARY NAMES pthread pthreadGC2 pthreadGC)` does not work on regular MingW installs but `find_package(Threads REQUIRED)` is pretty standard and works nicely, you can always try it out itself. I looked through the history of old `find_library` statement and it seems the story slowly evolved to this line from people using bigger build systems wrapping SRT building. Rules may be different there but I think `find_package()` should work. I I understand the concern for breaking those systems but I think now that we have MingW32 CI in place this will need to be re-adjusted so that it works for regular MingW32 install *and* third-party build systems, whatever they might be.

Reply:
Good, then I think we can risk breaking those other systems. Maybe this will need a fix in future, but this time the code could be kept cleaner.